config/v1/types_ingress.go

```diff
@@ -41,13 +43,128 @@ type IngressSpec struct {
+	// entry in the list of status.componentRoutes if the route is to be customized.
+	// +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=63
+	// +kubebuilder:validation:Required
+	// +required
+	Namespace string `json:"namespace"`
+
+	// name is the logical name of the route to customize.
+	//
+	// The namespace and name of this componentRoute must match a corresponding
+	// entry in the list of status.componentRoutes if the route is to be customized.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=256
+	// +kubebuilder:validation:Required
+	// +required
+	Name string `json:"name"`
+
+	// hostname is the hostname that should be used by the route.
+	// +kubebuilder:validation:Required
+	// +required
+	Hostname hostname `json:"hostname"`
+
+	// servingCertKeyPairSecret is a reference to a secret of type `kubernetes.io/tls` in the openshift-config namespace.
+	// The serving cert/key pair must match and will be used by the operator to fulfill the intent of serving with this name.
+	// +optional
+	ServingCertKeyPairSecret SecretNameReference `json:"servingCertKeyPairSecret"`
+}
+
+// ComponentRouteStatus contains information allowing configuration of a route's hostname and serving certificate.
+type ComponentRouteStatus struct {
+	// namespace is the namespace of the route to customize. It must be a real namespace. Using an actual namespace
+	// ensures that no two components will conflict and the same component can be installed multiple times.
+	//
+	// The namespace and name of this componentRoute must match a corresponding
+	// entry in the list of spec.componentRoutes if the route is to be customized.
+	// +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=63
+	// +kubebuilder:validation:Required
+	// +required
+	Namespace string `json:"namespace"`
+
+	// name is the logical name of the route to customize. It does not have to be the actual name of a route resource
+	// but it cannot be renamed.
+	//
+	// The namespace and name of this componentRoute must match a corresponding
+	// entry in the list of spec.componentRoutes if the route is to be customized.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=256
+	// +kubebuilder:validation:Required
+	// +required
+	Name string `json:"name"`
+
+	// defaultHostname is the hostname of this route prior to customization.
+	// +kubebuilder:validation:Required
+	// +required
+	DefaultHostname hostname `json:"defaultHostname"`
+
+	// consumingUsers is a slice of ServiceAccounts that need to have read permission on the servingCertKeyPairSecret secret.
+	// +kubebuilder:validation:MaxItems=5
+	// +optional
+	ConsumingUsers []consumingUser `json:"consumingUsers,omitempty"`
 
+	// currentHostnames is the list of current names used by the route.
```

Comment:
this is still confusing and maybe underspecified? What is expected to be here?

Reply:
`currentHostnames` should be the set of hostnames that are currently accepting and routing traffic by the route.

Reply:
why are there multiple hostnames?
what are the additional hostnames?
who provides them?
how can I change them as an admin?
what am I expected as an operator author?

Reply:
>why are there multiple hostnames?
what are the additional hostnames?

Given @deads2k [response](https://github.com/openshift/api/pull/852#issuecomment-793992844) it seems entirely valid to have multiple hostnames supported by routes, but a cluster-admin may only specify a single custom route.

>  who provides them?

The operator.

> how can I change them as an admin?

The initial design does not provide cluster admins with the means to disable default hostnames. This could be implemented in a future update by adding a boolean field communicating whether or not to disable defaults routes, unless this is a must-have feature.

> what am I expected as an operator author?

The operator author could configure their operator to continue to support the default route as the console team did or remove the default. I plan to update the existing enhancement and will add this information there.